lib/modules/dosomething/dosomething_user/dosomething_user.badge.inc

```diff
@@ -24,24 +35,72 @@ function dosomething_user_get_reportbacks_badge_images() {
 }
 
 function dosomething_user_get_kudos_badge_images() {
-  $badges[1] = '&#x1F6B2';
-  $badges[3] = '&#x1F6B2';
-  $badges[5] = '&#x1F6B2';
-  $badges[10] = '&#x1F6B2';
-  $badges[15] = '&#x1F6B2';
-  $badges[30] = '&#x1F6B2';
+  $badges[1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges[3] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges[5] = PARANEUE_PATH . '/images/badges/kudos-5.png';
+  $badges[10] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges[15] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges[30] = PARANEUE_PATH . '/images/badges/default.png';
 
   return $badges;
 }
 
+/**
+ * Gets all badges images for a category or subcategory of badges
+ *
+ * @param  $type
+ * @return [array]
+ */
+function dosomething_user_get_cause_badge_images($type) {
+  $badges['mental_health'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['animals'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['bullying'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['disasters'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['discrimination'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['education'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['environment'][1] = PARANEUE_PATH . '/images/badges/environment-1.png';
+  $badges['environment'][3] = PARANEUE_PATH . '/images/badges/environment-5.png';
+  $badges['homelessness'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['physical_health'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['poverty'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['relationships'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['sex'][1] = PARANEUE_PATH . '/images/badges/default.png';
+  $badges['violence'][1] = PARANEUE_PATH . '/images/badges/default.png';
+
+  return $badges[$type];
+}
+
+function dosomething_user_get_special_badge_images($specialty) {
+  $badges[$specialty][1] = PARANEUE_PATH . '/images/badges/special-1.png';
 
-function dosomething_user_get_badge_images($type) {
+  return $badges[$specialty];
+}
 
-  $handler = 'dosomething_user_get_' . $type . '_badge_images';
-  if (function_exists($handler)) {
-    $badges[$type] = $handler();
+/**
+ * Gets all badge images for a category or subcategory of badges
+ *
+ * @param  $type
+ * @param  $group
+ * @return [array]
+ */
+function dosomething_user_get_badge_images($type, $group) {
+  // If the badge type is part of a group (like cause area, action type, or special campaign),
+  // we call the function for that group to get the image array for the given subcategory
+  if ($group) {
```

Comment:
this isn't needed for merge, but it might be good to always call `dosomething_user_get_type_badge_images` and then only pass group if it's set